File: lib/genetic_algorithm/population/managers/CrossoverManager.py
from lib.genetic_algorithm.crossover_functions.CrossoverFunction import CrossoverFunction


class CrossoverManager():
    """
    Manager class for crossover methods
    """

    def __init__(self):
        self.crossover_fun = CrossoverFunction()

    def cross_over(self, individual1, individual2):
        """
        applies the crossover function to the two individuals and returns a new individual
        :param individual1: <InputIndividual> the first individual to do crossover
        :param individual2: <InputIndividual> the second individual to do crossover
        :return: <InputIndividual> the resulting individual to do crossover with the current crossover function
        """
        return self.crossover_fun.cross_over(individual1, individual2)

    def set_crossover_fun(self, new_crossover_fun):
        """
        setter
        """
        self.crossover_fun = new_crossover_fun

File: love_ga_wrapper/love_ga_wrapper/lib/genetic_algorithms/population/managers/CrossoverManager.lua
local class = require("love_ga_wrapper.lib.classes.class")
local extend = require("love_ga_wrapper.lib.classes.extend")
local CrossoverFunction =  require("love_ga_wrapper.lib.genetic_algorithms.crossover_functions.CrossoverFunction")
--------------------------------------------------------------------------------------------------------

-- class: CrossoverManager
-- Manager class for crossover methods
local CrossoverManager = class(function(self)
    self.crossover_fun = CrossoverFunction.new()
end)

-- crossover: Individual, Individual -> Individual
-- applies the crossover function to the two individuals and returns a new individual
function CrossoverManager.crossOver(self, individual1, individual2)
    return self.crossover_fun:crossover(individual1, individual2)
end

-- setter
function CrossoverManager.serCrossoverFun(self, new_crossover_fun)
    self.crossover_fun = new_crossover_fun
end

return CrossoverManager
